.github/workflows/MPI.yml

```
@@ -0,0 +1,258 @@
+# Perform builds with supported MPI versions
+name: MPI
+
+on:
+  # On pull_requests which mutate this CI workflow, using pr rather than push:branch to ensure that the merged state will be OK, as that is what is important here.
+  pull_request:
+    paths:
+      - ".github/workflows/MPI.yml"
+      - "tests/test_cases/simulation/test_mpi_ensemble.cu"
+      - "include/flamegpu/simulation/detail/MPISimRunner.h"
+      - "include/flamegpu/simulation/detail/AbstractSimRunner.h"
+      - "src/flamegpu/simulation/detail/MPISimRunner.cu"
+      - "src/flamegpu/simulation/detail/AbstractSimRunner.cu"
+      - "src/flamegpu/simulation/CUDAEnsemble.cu"
+      - "**mpi**"
+      - "**MPI**"
+  # Or trigger on manual dispatch. 
+  workflow_dispatch:
+
+defaults:
+  run:
+    # Default to using bash regardless of OS unless otherwise specified.
+    shell: bash
+
+# @todo - add a job/step which validates the version is of the correct format (for pre releases) and that the git tag matches the CMake/header version(s) 
+
+# Several jobs with some dependencies between them.
+# + Thorough Ubuntu builds
+#   + MPI 2.x+
+
+jobs:
+  build-ubuntu:
+    runs-on: ${{ matrix.cudacxx.os }}
+    strategy:
+      fail-fast: false
+      # Multiplicative build matrix
+      # optional exclude: can be partial, include: must be specific
+      matrix:
+        # CUDA_ARCH values are reduced compared to wheels due to CI memory issues while compiling the test suite.
+        cudacxx:
+          - cuda: "12.0"
+            cuda_arch: "50-real;90-real;90-virtual;"
```

For this CI we can just use single arch, not min max like in the thorough build to speed it up. 
```suggestion
            cuda_arch: "50-real;"
```